spock-core/src/main/java/org/spockframework/runtime/PlatformParameterizedSpecRunner.java

```diff
@@ -39,231 +39,38 @@ void runParameterizedFeature(SpockExecutionContext context, ParameterizedFeature
-        break;
-      }
-
-      dataProviders[i] = provider;
-    }
-
-    return dataProviders;
-  }
-
-  private Object[] getPreviousDataTableProviders(List<String> dataProviderVariables, Object[] dataProviders,
-                                                 DataProviderInfo dataProviderInfo) {
-    List<Object> result = new ArrayList<>();
-    previousDataTableVariablesLoop:
-    for (String previousDataTableVariable : dataProviderInfo.getPreviousDataTableVariables()) {
-      for (int i = 0, size = dataProviderVariables.size(); i < size; i++) {
-        String dataProviderVariable = dataProviderVariables.get(i);
-        if (previousDataTableVariable.equals(dataProviderVariable)) {
-          result.add(dataProviders[i]);
-          continue previousDataTableVariablesLoop;
-        }
-      }
-      throw new IllegalStateException(String.format("Variable name not defined (%s not in %s)!",
-          previousDataTableVariable, dataProviderVariables));
-    }
-    return result.toArray();
-  }
-
-  private Iterator[] createIterators(SpockExecutionContext context, Object[] dataProviders) {
-    if (context.getErrorInfoCollector().hasErrors()) {
-      return null;
-    }
-
-    Iterator[] iterators = new Iterator<?>[dataProviders.length];
-    for (int i = 0; i < dataProviders.length; i++)
-      try {
-        Iterator<?> iter = GroovyRuntimeUtil.asIterator(dataProviders[i]);
-        if (iter == null) {
-          supervisor.error(context.getErrorInfoCollector(), new ErrorInfo(context.getCurrentFeature().getDataProviders().get(i).getDataProviderMethod(),
-              new SpockExecutionException("Data provider's iterator() method returned null")));
-          return null;
-        }
-        iterators[i] = iter;
-      } catch (Throwable t) {
-        supervisor.error(context.getErrorInfoCollector(), new ErrorInfo(context.getCurrentFeature().getDataProviders().get(i).getDataProviderMethod(), t));
-        return null;
-      }
-
-    return iterators;
-  }
-
-  // -1 => unknown
-  private int estimateNumIterations(SpockExecutionContext context, Object[] dataProviders) {
-    if (context.getErrorInfoCollector().hasErrors()) {
-      return -1;
-    }
-    if (dataProviders.length == 0) {
-      return 1;
-    }
-
-    int result = Integer.MAX_VALUE;
-    for (Object prov : dataProviders) {
-      if (prov instanceof Iterator)
-      // unbelievably, DGM provides a size() method for Iterators,
-      // although it is of course destructive (i.e. it exhausts the Iterator)
-      {
-        continue;
-      }
-
-      Object rawSize = GroovyRuntimeUtil.invokeMethodQuietly(prov, "size");
-      if (!(rawSize instanceof Number)) {
-        continue;
-      }
-
-      int size = ((Number)rawSize).intValue();
-      if (size < 0 || size >= result) {
-        continue;
-      }
-
-      result = size;
+    } catch (InterruptedException ie) {
+      throw ie;
+    } catch (Exception e) {
+      // DataIterator.close doesn't throw
```

Comment:
🤔 Even so, something elsee might be throwing an exception here so I think it would be better to wrap it in a `RuntimeException` rather than accidentally swallowing it.

Reply:
done [1c18e7c](https://github.com/spockframework/spock/pull/1479/commits/1c18e7c1e230a08fb207292436d8e892bb04851b)